Kubernetes/HOWTO-on-prem.md

```diff
@@ -0,0 +1,103 @@
+# Kubernetes On-Premises Deployment From Scratch -- Start to Finish #
+This guide will walk you through deploying *Kubernetes 1.8* on a Linux master and join two Windows nodes to it without a cloud provider.
+
+**Note**: The guide will assume a local working directory of `~/kube` for the Kubernetes setup. If you choose a different directory, just replace any references to that path.
+
+## Building the Binaries ##
+We will begin with a Linux master node and a Windows worker node on RS3+, both of which are assumed to be VMs.
+
+First, install all of the pre-requisites on the master:
+
+    # apt-get install git build-essentials docker.io
+
+### Building Kubernetes ###
+We will need to build the `kubelet` and `kubeproxy` binaries for Windows from scratch _cross-compiling from Linux_, so let's [set up a Go environment](https://golang.org/doc/install#install) for that, first. This guide assumes Go 1.9 and has not been tested with anything else. With your Go environment ready (don't forget to set your `$GOPATH`!), we can build the Windows binaries:
+
+```bash
+$ K8SREPO="github.com/madhanrm/kubernetes"
+$ go get -d $K8SREPO
+$ cd $GOPATH/src/$K8SREPO
+$ KUBE_BUILD_PLATFORM=windows/amd64 make WHAT=cmd/kubectl

```

Comment:
Can you put this on two lines; it's confusing
$ KUBE_BUILD_PLATFORM=windows/amd64
$ make WHAT=cmd/kubectl

Reply:
This syntax is pretty common on Linux when you only want the env variable to apply to the command being executed (in this case 'make'). If this is split across two lines, then the value of the env variable will persist for all future commands in that shell too. That might be ok, but just mentioning the side effect ;)

Reply:
Sure, I can clarify that in the instructions and just unset the variable at the end.

Reply:
Should be KUBE_BUILD_PLATFORMS (with an S)

Reply:
If there are two lines for KUBE_BUILD_PLATFORMS and make command, make sure to export KUBE_BUILD_PLATFORMS environment variable
export KUBE_BUILD_PLATFORMS=windows/amd64